src/MonoMod.ILHelpers/MonoMod.ILHelpers.il

```diff
@@ -31,33 +31,23 @@
         callvirt instance !0 class [FUNC_ASSEMBLY]System.Func`1<!!T>::Invoke()
         ret
     }
-
-    // This exists to work around the fact that on old Mono, the normal string pinning sequence is broken.
-    // See docs/RuntimeIssueNodes.md
-    .method public hidebysig static native int GetRawDataOffset(object obj, uint8& rawData) cil managed aggressiveinlining
+    
+    .method public hidebysig static !!T& ObjectAsRef<T>(object obj) cil managed aggressiveinlining
     {
-        .custom instance void System.Runtime.CompilerServices.NullableContextAttribute::.ctor(uint8) = (
-            01 00 01 00 00
-        )
-
-        .maxstack 8
-        .locals init (
-            object pinned str,
-            void*& ptrRef
+        .locals (
+            object pinned pin,
+            !!T*& ref
         )
-
-        ldarg.1
-        conv.u
-
-        ldarg.0
-        stloc str
-        ldloca str
-        conv.u
-        stloc ptrRef
-        ldloc ptrRef
+        
+        ldarg obj
+        stloc pin
+        
+        ldloca pin
         ldind.i

```

Comment:
IIRC I'd found that this doesn't work on the old Mono where we needed the leper. This case needs the `conv.u` and `stloc` to force a typecast, or it'll throw for invalid IL.